zooniverse/front-end-monorepo
# VV - UI + Styles

## Package
- lib-subject-viewers

## Describe your changes
- Add 4, 8, 16, 32, and 64 example to Storybook
- Add Histogram range slider
- Fix scroll bug
- Inline OrbitControls import
- Incorporate [Figma Styles](https://www.figma.com/design/HCox3z5YQhoNcni70HRcwF/Volumetric-Viewer?node-id=0-1&p=f&t=RZUlnl7i3YGo9DGx-0)

## How to Review
- Run `lib-subject-viewers` Storybook locally
- Inspect styles in the deployed [project-branch](https://fe-project-branch.preview.zooniverse.org/projects/kbmantha/neurotracerbrain/classify/workflow/27910)
- Alternatively, you can run the local [app-project](https://fe-project-branch.preview.zooniverse.org/projects/kbmantha/neurotracerbrain/classify/workflow/27910)

Please Review
- Are layout and color styles implemented correctly?
  - Test both Desktop and Mobile in `app-project` or on the branch deploy
- Does resizing the window still enable full annotating?
- Planes
  - Does the toggle to expand/collapse work?
  - Does the "forward 10" move forward 10 frames?
  - Does holding the "forward 10" move forward at an interval?
  - Does letting go of the "forward 10" stop moving forward at an interval?
  - Does adjusting the slider work?
  - Does clicking anywhere on the cube create+add to an the annotation?
  - Does holding shift+click on the cube create a new annotation?
- Cube
  - Can you rotate the cube?
  - Can you see annotations?
  - Does clicking anywhere on the cube create+add to an the annotation?
  - Does holding shift+click on the cube create a new annotation?
- Histogram
  - Do both the plane and the cube remove pixels/voxels when the histogram range is narrowed?
  - Does the Histogram drawing re-normalize when adjusting the sliders?

Design Review Notes (from meeting with Sean)
- Hide coordinate when collapsed
- Keep coordinate size smaller when expanded
- Start page with x slice open
- 20px padding on the cube
- Remove rounded corners on cube
- Remove stroke from dark mode
- Cut out the bottom black from the histogram
- Click and hold to “loop through”
- When the histogram and axes touch is the min size for the cube
    - When it gets to that size, then move to stacked layout
- When threshold is moved, sometimes the annotations are not redrawn and remain hidden until the frame is changed

Future PR / Not Implemented
- Ability to delete an Annotation in the cube
- Ability to "undo" in the Toolbar
- Get the outline to travel with the frame (math is complicated)

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook
